gtest/src/manager.rs

```diff
@@ -948,6 +948,40 @@ impl ExtManager {
 
         core_processor::handle_journal(journal, self);
     }
+
+    fn remove_reservation(&mut self, id: ProgramId, reservation: ReservationId) -> Option<()> {
+        let res = self.adjust_genuine_program(id, |genuine_program| {
+            let genuine_program = genuine_program.expect("no genuine program was found");
+            genuine_program
+                .gas_reservation_map
+                .remove(&reservation)
+                .map(|_| ())
+                .or_else(|| {
+                    log::error!("Tried to remove non-existent reservation");
+
+                    None
+                })
+        });
+
+        res.map(|_| {
+            self.gas_tree
+                .consume(reservation)
+                .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
+        })
+    }
```

Comment:
It might be more clear to use a simple `bool` as the return value instead of `Option<()>`. 
Additionally, the map(|_| ..) logic seems a bit complex; perhaps consider using bools for better clarity.

Reply:
Or consider to merge blocks and use `if let Some(_reservation)`

Reply:
> Or consider to merge blocks and use `if let Some(_reservation)`

unfortunately can't do that, as `adjust_genuine_program` actually has a mutable access to the self, while `self.gas_tree` ops - immutable, can't do that.

Reply:
> It might be more clear to use a simple `bool` as the return value instead of `Option<()>`. Additionally, the `map(|_| ...)` logic seems a bit complex; perhaps consider using bools for better clarity.

changed in favor of bool